### PR TITLE
KWS-2655:  Set up UI lib to send events

### PR DESCRIPTION
**Pull request by Tbruni85**
## JIRA
https://jira.superawesome.tv/browse/KWS-2655    

## Description
Set up the UI un order to allow single views to send events

## Screenshots
n/a

### CODE REVIEW

File: PopJamUI/Classes/Utilis/EventConstructor.swift [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+//  EventConstructor.swift
+//  PopJamUI
+//
+//  Created by Tiziano Bruni on 02/05/2018.
+//
+
+import Foundation
+import SAProtobufs
+
+internal struct EventStruct: SocialStatsModelProtocol {
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/Utilis/EventConstructor.swift]: Maybe name this `StatModel` or something similar? Or just `Stat`. Below you have `Session` and the naming works ok.

File: PopJamUI/Classes/FeedView/DataProvider/BaseFeedDataProvider.swift [hunk 2]
```diff
@@ -233,4 +236,14 @@ public class BaseFeedDataProvider: DataProvider {
                          error: self.error)
         }
     }
+    
+    // MARK: Send event method
+    
+    public func sendEvent(event: SocialStatsModelProtocol) {
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/FeedView/DataProvider/BaseFeedDataProvider.swift]: This exact `sendEvent` method will be in all Data Providers, copy-pasted. 
Any way to group this logic into a common DataProvider or StatsManager class, etc?

File: Example/PopJamUI/ViewControllers/ViewController.swift [hunk 3]
```diff
@@ -225,10 +228,13 @@ class ViewController: UIViewController,
...
         let actionService = serviceProvider.getService(withType: UserSocialActionsServiceProtocol.self)!
+        let eventsService = ServiceProvider.shared.getServices(type: .eventsService) as! StatsServiceProtocol
+        
         let dataProvider = FeedItemsDataProvider(feedItemsService: feedItemsService,
                                                  actionService: actionService,
-                                                 loggedUser: fakeLoggedUser)
-        let feedItemsViewModel = FeedItemsViewModel(withDataProvider: dataProvider)
+                                                 loggedUser: fakeLoggedUser,
+                                                 eventsService: eventsService)
+        let feedItemsViewModel = FeedItemsViewModel(withDataProvider: dataProvider, uuid: ServiceProvider.shared.sessionUUID)
```
Comment (devgabrielcoman) [inline Example/PopJamUI/ViewControllers/ViewController.swift]: Don't particularly like that the sessionUUID is passed in the VideModel like that. 
Kinda thing that external users shouldn't really care about SessionUUID and so on.
And when deploying this SDK to the NBCU / Vanilla / PJ 2 / BattleJam apps, can't see why we should care about it either. 

What we did on Android is ex added a bunch of new methods to SessionServiceProtocol (and its implementation in the Data SDK) to allow us to generate, save, obtain, etc one of these UUIDs on the fly through some method calls.
What you end up doing then is injecting the SessionServiceProtocol in the DataProvider, and letting the DataProvider internally know about UUIDs and how to get them, when to refresh them, etc.

File: PopJamUI/Classes/FeedView/DataProvider/BaseFeedDataProvider.swift (hunk 2 above)
Comment (Tbruni85) [reply PopJamUI/Classes/FeedView/DataProvider/BaseFeedDataProvider.swift]: it's in the BaseFeedDataProvider exactly for that reason, I'll create a new BaseDataProvider to keep all these shared functions

File: PopJamUI/Classes/Utilis/EventConstructor.swift (hunk 1 above)
Comment (Tbruni85) [reply PopJamUI/Classes/Utilis/EventConstructor.swift]: `Stat` sounds good to me too! 👍

File: Example/PopJamUI/ViewControllers/ViewController.swift (hunk 3 above)
Comment (Tbruni85) [reply Example/PopJamUI/ViewControllers/ViewController.swift]: We can move the sessionUUID inside the Util file.
Anyway the App will need to have access to it cause a new UUID must be generated at 
* Login
* Logout
* App back to foreground

Which are event triggered on the App side

File: PopJamUI/Classes/FeedView/DataProvider/BaseFeedDataProvider.swift (hunk 2 above)
Comment (devgabrielcoman) [reply PopJamUI/Classes/FeedView/DataProvider/BaseFeedDataProvider.swift]: Kewl 👍

File: Example/PopJamUI/ViewControllers/ViewController.swift (hunk 3 above)
Comment (devgabrielcoman) [reply Example/PopJamUI/ViewControllers/ViewController.swift]: Hmm, are we sure that app back to foreground requires a new UUID?

Comment (me2k) [reply Example/PopJamUI/ViewControllers/ViewController.swift]: Can we not create a manager in the Data SDK that provides the UUID?

**Pull request merged**